### PR TITLE
[GitLab config] Include token by default + nits

### PR DESCRIPTION
**Pull request by adamboros**
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [ ] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs

The GitLab MR comments page has a snippet example. I've edited the example to be more concise and redirect users to the sample CI for more details. This example should be about how to use the $PAT part as we should only expect users here who already have an App - CI setup.

I want to propose to explicitly use `SEMGREP_APP_TOKEN: $SEMGREP_APP_TOKEN` in configs provided by us: it's clearer to the user that
- a token is expected, 
- it can be renamed,
- and when debugging is needed it helps reading the file.

Semgrep retrieving the token in the background is a magic black box, which is nice, but we shouldn't encourage it. 

Also fixed a link and reformatted a few bits.

### CODE REVIEW

**Review by adamkvitek — COMMENTED** [review]
@adamboros thank you, made a quick review. Great additions. Please check.

File: docs/semgrep-ci/sample-ci-configs.md [hunk 1]
```diff
@@ -147,16 +147,16 @@ semgrep:
...
-    # Option 1: Scan with rules set in Semgrep App's rule board
+    # [Option 1] Scan with rules set in Semgrep App's rule board
     # Get your token at semgrep.dev/orgs/-/settings/tokens.
     SEMGREP_APP_TOKEN: $SEMGREP_APP_TOKEN
-    # Option 2: set hard-coded rulesets, viewable in logs.
+    # [Option 2] set hard-coded rulesets, viewable in logs.
     # SEMGREP_RULES: p/default # See more at semgrep.dev/explore.
 
-  # == Other optional settings in the `variables:` block
+  # == Other optional settings in the `variables:` block ==
```
[the remark this answers is not in the document]
Comment (adamboros) [reply docs/semgrep-ci/sample-ci-configs.md:156]: This is a note and user shouldn't uncomment it. The `==` here is for showing that this is a header, I've added it to the end for emphasis (makes it easier to find when skimming through).

File: docs/semgrep-app/notifications.md [hunk 2]
```diff
@@ -84,24 +84,24 @@ To enable MR comments:
 5. Click to **Add variable**, give the new variable the key `PAT` and use the token you copied in step 3 as the value. And then, select **mask variable** and **UNSELECT "protect variable"**.
 6. Update your .gitlab-ci.yml file with variable `GITLAB_TOKEN` and value `$PAT`. See the example code below for details:
 
-For example:
+#### Example
```
[the remark this answers is not in the document]
Comment (adamboros) [reply docs/semgrep-app/notifications.md:87]: @adamkvitek I'm a little adamant about this, since the example is about where $PAT should be and not with the purpose of providing a sample/example CI code to copy. Wdyt?

File: docs/semgrep-app/notifications.md [hunk 3]
```diff
@@ -84,24 +84,24 @@ To enable MR comments:
 5. Click to **Add variable**, give the new variable the key `PAT` and use the token you copied in step 3 as the value. And then, select **mask variable** and **UNSELECT "protect variable"**.
 6. Update your .gitlab-ci.yml file with variable `GITLAB_TOKEN` and value `$PAT`. See the example code below for details:
```
[the remark this answers is not in the document]
Comment (adamkvitek) [reply docs/semgrep-app/notifications.md:85]: Note: change this suggestion depending on the header changed in: https://github.com/returntocorp/semgrep-docs/pull/573/files/b6b471393e2fccc4e2d230bb9b005c335b392e47#r897781778

File: docs/semgrep-app/notifications.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (adamkvitek) [reply docs/semgrep-app/notifications.md:87]: @adamboros makes sense! 

We have two options: 

1. Make the header more specific to what you want to convey here. What about (feel free to change this one):
    ```md
    #### $PAT variable example
    ```

2. (Which I find much better.) I think we do not need a separate header for something that can be explained only by indentation. This will make clear that the example belongs to step number 6. Making a separate header is a bit too much here I think. See suggestion below the original:

Original:
```md
6. Update your .gitlab-ci.yml file with variable `GITLAB_TOKEN` and value `$PAT`. See the example code below for details:

#### Example
```yaml
semgrep:
  image: returntocorp/semgrep
  script:
    - semgrep ci
  rules:
  - if: $CI_MERGE_REQUEST_IID

  variables:
    SEMGREP_APP_TOKEN: $SEMGREP_APP_TOKEN
    GITLAB_TOKEN: $PAT
```
```

Suggesting
```md
6. Update your .gitlab-ci.yml file with variable `GITLAB_TOKEN` and value `$PAT`. See the example code below for details:
    ```yaml
    semgrep:
      image: returntocorp/semgrep
      script:
        - semgrep ci
      rules:
      - if: $CI_MERGE_REQUEST_IID

      variables:
        SEMGREP_APP_TOKEN: $SEMGREP_APP_TOKEN
        GITLAB_TOKEN: $PAT
    ```
```

[the remark this answers is not in the document]
Comment (adamkvitek) [reply docs/semgrep-app/notifications.md:87]: (Note) There is some weird issue with formatting here, please consider `Suggesting:` as not being a part of the code snippet above. @adamboros

[the remark this answers is not in the document]
Comment (adamboros) [reply docs/semgrep-app/notifications.md:87]: @adamkvitek I don't think this results in any indentation difference once rendered, am I missing something?

File: docs/semgrep-app/notifications.md [hunk 4]
```diff
@@ -75,7 +75,7 @@ Automated comments on GitLab merge requests are displayed as follows:
 <img width="600" src="/docs/img/gitlab-mr-comment.png" alt="Screenshot of a GitLab MR comment" /><br />
 An inline GitLab merge request comment left by a custom Semgrep rule
 
-To enable MR comments:
+#### To enable MR comments
```
[the remark this answers is not in the document]
Comment (underyx) [reply docs/semgrep-app/notifications.md:78]: Calling it a "procedure" sounds super official. `#### Enabling GitLab MR comments` would look nicer in my opinion.

File: docs/semgrep-app/notifications.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (adamkvitek) [reply docs/semgrep-app/notifications.md:87]: @adamboros If you use VS code, there are plugins to build preview inside of the editor, which can help with some formatting things, although usually, you need to build the docs as well to make sure the output will work. ;-)

File: docs/semgrep-app/notifications.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (adamkvitek) [reply docs/semgrep-app/notifications.md:78]: @underyx look at line 69. There is a problem. We cannot have two so much similar headers. Please, read my comment above. 😅

[the remark this answers is not in the document]
Comment (adamkvitek) [reply docs/semgrep-app/notifications.md:78]: Maybe I am just too wordy for you. New suggestion. :D 
```suggestion
To enable GitLab merge request comments, follow these steps: 
```

**Comment by adamkvitek** [comment]
Hi there @adamboros, any update on this PR? :-)

**Comment by adamkvitek** [comment]
@adamboros any update please? :-) Let me know if there is anything we can do to speed up merging your update.

File: docs/semgrep-app/notifications.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (adamkvitek) [reply docs/semgrep-app/notifications.md:92]: Updating in the document.

**Review by adamkvitek — APPROVED** [review]
